### ISSUE TITLE
[Check Aggregates]: monitoring a collection of machines running a given service and alarming on a set of criteria.

### ISSUE THREAD

**Issue by chiriaev** [sensu/sensu#803]
Sean, Hi. What do I/we need to do to make the below possible? Regards, Andrei
...
1- Four physical servers hosting all the monitoring stack are at play. (i.e.: later, "head-ends")
2- The community's "check-aggregate.rb" is running on all head-ends.
3- The server, master in fact, is publishing an aggregate check request.
4- The clients run the check and and return results. Let's say there are 10 clients that published results back.
5- The server(s) do their magic in Redis. (i.e.: every Sensu server may pick-up the check's output)

Now, all the conditions are met to raise an alarm.

a- How do I make sure that the check in "2-" is executed only once? I want the solution to be resilient if one head-end goes down.
b- How would I create an extension/handler based on "2-"'s logic? It should only be executed by the master. How do I deal with point "5-"? (i.e.: when a request gets to a slave, can't even send it to the "master"; there is no server socket or dedicated queue in place)

The “aggregates” feature feels to be popular in different circles. It something that unfortunately doesn’t have a production grade implementation. Let’s try to work it out.
...

**Comment by calebhailey** [comment]
Tagged this in a new meta issue to discuss an aggregates redux. Would love comments on #1218

**Comment by calebhailey** [comment]
A proposal for "aggregates 2.0" has been published as a comment on issue #1218. Barring any objections, this proposal will be implemented in Sensu 0.24.

**Comment by solarkennedy** [comment]
No objections from me.

**Comment by portertech** [comment]
Aggregates 2.0 has been merged into the release/0.24 branch #1273 Sensu 0.24 will be available sometime next week.

**Comment by portertech** [comment]
Going to close this issue in light of 0.24 and Aggregates 2.0. Discussion will likely change as we have a new(ish) primitive to play with.
